core/spec/models/spree/stock/estimator_spec.rb

```diff
@@ -140,6 +140,36 @@ module Stock
             expect(shipping_rates.first.tax_rate).to eq(tax_rate)
           end
         end
+
+        it 'uses the configured shipping rate selector' do
+          shipping_rate = Spree::ShippingRate.new
+          allow(Spree::ShippingRate).to receive(:new).and_return(shipping_rate)
+
+          selector_class = Class.new do
+            def initialize(_); end;
+
+            def find_default
+              Spree::ShippingRate.new
+            end
+          end
+          Spree::Config[:shipping_rate_selector_class] = selector_class
+
+          subject.shipping_rates(package)
+
+          expect(shipping_rate.selected).to eq(true)
```

Comment:
Updated again to return a real `ShippingRate` to verify state instead of behavior. :)